src/serverModel.ts

```diff
@@ -153,28 +304,37 @@ export default function ServerModel(mpInstance) {
             };
         }
 
-        return jsonObject;
+        return jsonObject as IConsentStateDTO;
     };
 
-    this.createEventObject = function(event, user) {
-        var uploadObject = {};
-        var eventObject = {};
+    this.createEventObject = function(
+        event: BaseEvent,
+        user?: MParticleUser
+    ): SDKEvent | IUploadObject {
+        var uploadObject: Partial<IUploadObject> = {};
+        var eventObject: Partial<SDKEvent> = {};
 
+        // TODO: What is the purpose of setting OptOut to the inverse of the Store's
+        //       isEnabled property and not just `false`?
         var optOut =
             event.messageType === Types.MessageType.OptOut
                 ? !mpInstance._Store.isEnabled
                 : null;
 
+        // TODO: Why is Webview Bridge Enabled or Opt Out necessary here?
```

Comment:
I believe if web view bridge is enabled, there may not be a session id because the event is associated with the native sdk's  session.  I would have to dig in more to remember why optout is included here.